packages/web/src/components/buy-sell-modal/hooks/useTokenBalanceManager.ts

```diff
@@ -0,0 +1,123 @@
+import { useMemo } from 'react'
+
+import {
+  useAudioBalance,
+  useTokenBalance,
+  getTokenMintName
+} from '@audius/common/api'
+import { Status } from '@audius/common/models'
+import { TokenInfo } from '@audius/common/store'
+import { isNullOrUndefined } from '@audius/common/utils'
+import { AUDIO, FixedDecimal } from '@audius/fixed-decimal'
+
+import type { BalanceConfig } from '../types'
+
+export type TokenBalanceManager = {
+  inputBalance: BalanceConfig
+  outputBalance: BalanceConfig
+}
+
+/**
+ * Shared hook for managing token balance fetching and formatting
+ * Used by both ConvertTab and SellTab to eliminate code duplication
+ */
+export const useTokenBalanceManager = (
+  inputToken: TokenInfo,
+  outputToken: TokenInfo
+): TokenBalanceManager => {
+  // Fetch AUDIO balance for current user
+  const { accountBalance } = useAudioBalance({ includeConnectedWallets: false })
+
+  // Fetch balance for input token
+  const { data: inputTokenBalanceData, status: inputTokenBalanceStatus } =
+    useTokenBalance({
+      token: getTokenMintName(inputToken.symbol)
+    })
+
+  // Fetch balance for output token
+  const { data: outputTokenBalanceData, status: outputTokenBalanceStatus } =
+    useTokenBalance({
+      token: getTokenMintName(outputToken.symbol)
+    })
+
+  // Determine loading state for input token
+  const isInputBalanceLoading = useMemo(() => {
+    if (inputToken.symbol === 'AUDIO') {
```

Comment:
def a bit confusing with using he "useAudioBalance" for input/output if it happens to be it, can we not have useTokenBalance cover AUDIO?